### ISSUE TITLE
Removing Bloat, Specifically Redux?

### ISSUE THREAD

**Issue by tripflex** [No3x/wp-mail-logging#36]
Any specific reason for using Redux framework in the plugin?  It adds a lot of bloat and while debugging my sites I find numerous filters, actions, updates,etc, made by redux when really it shouldn't be doing anything on those pages ... basically just wondering, why add bloated framework when there are only a few simple settings?

**Comment by No3x** [comment]
Hi @tripflex, please see https://wordpress.org/support/topic/is-redux-framework-really-necessary

**Comment by tripflex** [comment]
Yeah I agree, that's my main reason for asking .... I pretty much build all my settings, help tabs, etc, from scratch and can probably help you to remove Redux and move to a simple solution using core handling.

Is there anything specific I need to know about redux, your integration of it, or anything specific that relies on Redux in the codebase?

**Comment by No3x** [comment]
Redux uses a global variable. The usage of this global is [scattered](https://github.com/No3x/wp-mail-logging/search?utf8=%E2%9C%93&q=%24wpml_settings) above all components. A good start to register is https://github.com/No3x/wp-mail-logging/blob/master/WPML_Init.php. Once registered addActionsAndFilters is called in your class (see https://github.com/No3x/wp-mail-logging/blob/514c9b28f43ce5dcc55465eccc9d92790c37c89a/WPML_LogRotation.php#L25-L33) - it runs very earls so you can add ordinary hooks.

My attempt to integrate redux clean failed at https://github.com/No3x/wp-mail-logging/blob/514c9b28f43ce5dcc55465eccc9d92790c37c89a/WPML_OptionsManager.php#L28-L46 because this settings werent available at every point in execution time and child/parent class.

**Comment by No3x** [comment]
Any progress on this?

**Comment by tripflex** [comment]
Nah haven't had a chance to do anything with it yet, maybe sometime next week.  When I do i'll fork it and add a PR with checklist of items to complete

**Comment by mkuplens** [comment]
If you'll excuse my chiming in, just a few thoughts on this to offer:

It's well worth eliminating Redux if possible. It may offer some shortcuts but it's adding a whole lot of heft and a pretty rough end-user experience.

At first install of your plugin, Redux hijacks the admin with its "welcome to Redux Framework" screen (my reaction: "What the heck is Redux Framework? I didn't install/want/ask for that!"), leaving me to dig through the admin to try and find the admin for the thing I wanted and start thinking about uninstalling/scanning my WordPress installation for exploits.

Once I've found your settings screen, I was immediately bombarded with promo messages from Redux (e.g., http://cl.ly/gByu, http://cl.ly/gC8a), which obscured your actual settings and made me less likely to use the tool.

Looking through the settings panel, I was a bit confused as to why you'd need their admin framework in the first place. The bells and whistles it provides seem rather overkill for your plugin.

Given the limited number of settings, import/export is unlikely to be particularly valuable; it's not as if there's a half-hour setup that needs to be recovered/restored. It's also not clear what date/time format would be used instead of WordPress's; could that be removed?

Looking at the auto-delete settings: sliders can be useful with stops, but as a generic number incrementor/decrementor, they don't add much value and actually make it harder than simple key-entry to hit a precise value. Switching those to simple text inputs would also allow the toggles to be removed ("Leave blank to disable").

Finally, the role selector is ok, but doesn't reflect the list of roles available to select in the user editor. Most WordPress users will be unfamiliar with the under-the-hood admin permission system.

Feel free to ignore or shoot me an email at m@mkuplens.com if you'd like to chat further!

Cheers,

-Martin

**Comment by No3x** [comment]
@mkuplens I really appreciate your contribution. 

> At first install of your plugin, Redux hijacks the admin with its "welcome to Redux Framework" screen (my reaction: "What the heck is Redux Framework? I didn't install/want/ask for that!"), leaving me to dig through the admin to try and find the admin for the thing I wanted and start thinking about uninstalling/scanning my WordPress installation for exploits.

I want to refer to this review: [Nice plugin, but the Redux splash page is confusing](https://wordpress.org/support/topic/nice-plugin-but-the-redux-splash-page-is-confusing). This actually confirms your doubts about redux. But I want to mention that ads are shown with `WP_DEBUG` set `true` only. (this was the result of my testing for this issue). And I think this is actually fair for redux to show ads to developers.

> Looking at the auto-delete settings: sliders can be useful with stops, but as a generic number incrementor/decrementor, they don't add much value and actually make it harder than simple key-entry to hit a precise value. Switching those to simple text inputs would also allow the toggles to be removed ("Leave blank to disable").

I also want to refer to this issue at redux: [Slider with suggested step](https://github.com/reduxframework/redux-framework/issues/2189). 
I actually cared about some of issues to find a solution.

@tripflex actually announced to work on this. I'm not aware of the current state.

You & everyone are invited to join the chat [![Join the chat at https://gitter.im/No3x/wp-mail-logging](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/No3x/wp-mail-logging?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) 
I will edit this post as soon as I have some time to answer in detail.

**Comment by donmhico** [comment]
@tripflex @mkuplens - Thank you all for chiming in on this issue. I'm closing this issue as we've removed the Redux Framework in `1.11.0`.
